src/vm/threads.h

```diff
@@ -7039,6 +7010,8 @@ struct ThreadLocalInfo
     Thread* m_pThread;
     AppDomain* m_pAppDomain; // This field is read only by the SOS plugin to get the AppDomain
     void** m_EETlsData; // ClrTlsInfo::data
+    EventPipeBufferList* m_pThreadEventBufferList;
+    bool m_threadEventWriteInProgress;
```

Comment:
Fyi changes in DeAllocateBuffers have broken the original usage of m_threadEventWriteInProgress. It is dead-code at this point. Originally this field was used so that DeAllocateBuffers could clean up per-thread buffers that were still in use. Looking over the current code I believe we are leaking all the thread local buffers on shutdown.

I think we'd be OK checking in with the leaks and then fixing it in a follow-up PR.



Background: Prior to the PR we had two per-thread volatile fields and the code used them like this:

    /* cleanup-thread in DeAllocateBuffers() */
    if(!pThread->m_threadEventWriteInProgress)
    {
        // delete all the memory in the buffers
        m_pThreadEventBufferList = NULL;
    }
    else
    {
         // leak it
    }


    /* event writing thread in EventPipeBufferManager::WriteEvent */
    m_threadEventWriteInProgress = TRUE 
    pBufferList = m_pThreadEventBufferList
    WriteToTheBuffer(pBufferList)
    m_threadEventWriteInProgress = FALSE

We've since eliminated that code in DeAllocateBuffers() and I didn't notice any replacement. 

That original code didn't look thread-safe either. The WriteEvent thread could set the in-use flag and begin using the buffers just after the cleanup thread checked they weren't in use.